File: frontend/lib/icons.ts
// Placeholder icon set — slugs mirror the backend ICON_POOL. Swap emoji for
// real icons later without touching logic. Keep in sync with core/icons.py.
export const ICON_EMOJI: Record<string, string> = {
  tree: "🌳",
  cat: "🐱",
  apple: "🍎",
  sun: "☀️",
  moon: "🌙",
  star: "⭐",
  dog: "🐶",
  fish: "🐟",
  bird: "🐦",
  leaf: "🍃",
  flower: "🌸",
  house: "🏠",
  car: "🚗",
  boat: "⛵",
  heart: "❤️",
  cloud: "☁️",
  rain: "🌧️",
  snow: "❄️",
  fire: "🔥",
  key: "🔑",
  book: "📖",
  ball: "⚽",
  cake: "🍰",
  bell: "🔔",
  guitar: "🎸",
  rocket: "🚀",
  crown: "👑",
  gift: "🎁",
  camera: "📷",
  clock: "⏰",
  umbrella: "☂️",
  balloon: "🎈",
  anchor: "⚓",
  diamond: "💎",
  mushroom: "🍄",
  cactus: "🌵",
  grapes: "🍇",
  lemon: "🍋",
  pizza: "🍕",
  hat: "🎩",
};

/** All selectable icon slugs, in a stable display order. */
export const ALL_ICONS = Object.keys(ICON_EMOJI);

export function emojiFor(slug: string): string {
  return ICON_EMOJI[slug] ?? "❔";
}

File: backend/app/core/icons.py
import random

# Placeholder icon slugs. Swap/extend with your real icon set later.
# With 24 icons and 3 ordered picks there are 24*23*22 = 12,144 unique combos.
ICON_POOL = [
    "tree", "cat", "apple", "sun", "moon", "star",
    "dog", "fish", "bird", "leaf", "flower", "house",
    "car", "boat", "heart", "cloud", "rain", "snow",
    "fire", "key", "book", "ball", "cake", "bell",
]

ICON_COUNT = 3


def random_icon_set() -> list[str]:
    """Return an ordered list of 3 distinct icon slugs."""
    return random.sample(ICON_POOL, ICON_COUNT)


def icons_to_password(icons: list[str]) -> str:
    """The default password is the icon slugs joined: 'tree_cat_apple'."""
    return "_".join(icons)
